pkg/apis/feature/features_test.go

```diff
@@ -56,3 +57,14 @@ func TestGetFlags(t *testing.T) {
 	require.True(t, flags.IsAllowed("my-allowed-flag"))
 	require.False(t, flags.IsAllowed("non-disabled-flag"))
 }
+
+func TestShouldNotOverrideDefaults(t *testing.T) {
+
+	f, err := NewFlagsConfigFromMap(map[string]string{})
+	require.Nil(t, err)
+	require.NotNil(t, f)
+
+	if !f.IsDisabled(KReferenceGroup) && !f.IsEnabled(KReferenceGroup) {
```

Comment:
> I've reviewed the part where we call NewFlagsConfigFromMap with an empty string map. If I understand correctly, this will return an empty flag. Wouldn't this potentially cause an issue at line 65? (nvm, it only checks for nil. The flag is empty, and it doesn't mean it is nil)

Exactly

> Also, I noted that with use of KReferenceGroup as a feature flag name. Is this specifically for testing purposes? Why we are using KReferenceGroup? And the if statement here is used to validate that KReferenceGroup only allow values from "Disabled" or "Enabled" right? How can that be related to "TestShouldNotOverrideDefaults"?

This assertion is the "regression test" for the original issue and if you remove the solution this assertion will fail, there is no specific reason for using that feature flag `KReferenceGroup`, another one would have worked as well